lib/assets.dart

```diff
@@ -0,0 +1,34 @@
+import 'dart:ui';
+
+import 'flame.dart';
+
+class ImageAsset {
```

Comment:
I think it might be even cleaner to have all constructors take an already loaded image.

then we can add async static factories constructors that take in the string. Like

either
```
final image = await Image.load("filename.png");
final sprite = Sprite(image, foo, bar);
```
or
```
final sprite = await Sprite.load("filename.png", foo, bar);
```

Reply:
we could get rid of unloaded stuff altogether if we let go of having a way to create a sprite w/o await